src/NuGet.Clients/NuGet.CommandLine/Commands/Command.cs

```diff
@@ -128,19 +130,25 @@ public void Execute()
         /// </summary>
         protected void SetDefaultCredentialProvider()
         {
-            // Register an additional provider for the console specific application so that the user
-            // will be prompted if a proxy is set and credentials are required
-            var credentialProvider = new SettingsCredentialProvider(
-                new ConsoleCredentialProvider(Console),
-                SourceProvider,
-                Console);
-            HttpClient.DefaultCredentialProvider = credentialProvider;
+            var providers = new List<Credentials.ICredentialProvider>();
+            var pluginProviders = new PluginCredentialProviderBuilder(Settings).BuildAll();
+
+            providers.Add(new CredentialProviderAdapter(new SettingsCredentialProvider(
+                NullCredentialProvider.Instance, SourceProvider, Console)));
```

Comment:
Let's please change the Setting provider not to get CredentialProvider that way we will not need to have the NullCredentialProvider